packages/cli/test/scripts/e2e_test_env.ts

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import path from "node:path";
+import {CLClient, ELClient} from "../utils/simulation/interfaces.js";
+import {SimulationEnvironment} from "../utils/simulation/SimulationEnvironment.js";
+import {getEstimatedTTD, logFilesDir} from "../utils/simulation/utils/index.js";
+import {connectAllNodes} from "../utils/simulation/utils/network.js";
+
+const secondsPerSlot = 4;
+const cliqueSealingPeriod = 5;
+const genesisDelaySeconds = 30 * secondsPerSlot;
+const altairForkEpoch = 1;
+const bellatrixForkEpoch = 2;
+const capellaForkEpoch = 3;
+// Make sure bellatrix started before TTD reach
+const additionalSlotsForTTD = 2;
+
+const ttd = getEstimatedTTD({
+  genesisDelaySeconds,
+  bellatrixForkEpoch,
+  secondsPerSlot,
+  cliqueSealingPeriod,
+  additionalSlots: additionalSlotsForTTD,
+});
+
+const env = await SimulationEnvironment.initWithDefaults(
+  {
+    id: "e2e-test-env",
+    logsDir: path.join(logFilesDir, "e2e-test-env"),
+    chainConfig: {
+      ALTAIR_FORK_EPOCH: altairForkEpoch,
+      BELLATRIX_FORK_EPOCH: bellatrixForkEpoch,
+      CAPELLA_FORK_EPOCH: capellaForkEpoch,
+      GENESIS_DELAY: genesisDelaySeconds,
+      TERMINAL_TOTAL_DIFFICULTY: ttd,
+    },
+  },
+  [
+    {id: "node-1", cl: CLClient.Lodestar, el: ELClient.Geth, keysCount: 32, mining: true},
+    {id: "node-2", cl: CLClient.Lodestar, el: ELClient.Nethermind, keysCount: 32},
+  ]
+);
+
+await env.start({runTimeoutMs: 0});
+await connectAllNodes(env.nodes);
```

Comment:
Does this approach leave a network alive for the entire duration of the tests? Which is going through forks? If yes, this means test outcomes of different e2e tests influence each other since they share global state, is that correct?

Reply:
Yes the network will be alive through out execution of **all e2e tests**. It's like connecting to external node, but with this the node is configured and running locally.

Every e2e tests should scaffold their specific network condition on this live network. e.g. If one test need to a contract, can deploy it. Or one network need a specific account, can transfer funds from genesis account. This way execution of tests would not effect each other.